client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -15,81 +15,67 @@ import TokenField from 'components/token-field';
 export default class ProductVariationTypesForm extends Component {
 
 	static propTypes = {
-		product: PropTypes.shape( {
-			id: PropTypes.number.isRequired,
-			name: PropTypes.string.isRequired,
-			type: PropTypes.string.isRequired,
-		} ),
-		variations: PropTypes.arrayOf( PropTypes.shape( {
-			type: PropTypes.string.isRequired,
-			values: PropTypes.arrayOf( PropTypes.string )
-		} ) ),
+		product: PropTypes.object.isRequired,
+		editProduct: PropTypes.func.isRequired,
 	};
 
 	constructor( props ) {
 		super( props );
 
-		this.state = {
-			variations: this.props.variations || this.getInitialFields(),
-		};
-
-		this.addVariation = this.addVariation.bind( this );
+		this.addVariationType = this.addVariationType.bind( this );
 		this.updateType = this.updateType.bind( this );
 		this.updateValues = this.updateValues.bind( this );
 	}
 
-	getInitialFields() {
-		return [ this.getNewFields() ];
-	}
-
 	getNewFields() {
 		return {
 			type: '',
 			values: [],
 		};
 	}
 
+	addVariationType() {
+		const updatedVariations = [ ...this.props.product.variationTypes, this.getNewFields() ];
+		this.props.editProduct( this.props.product.id, 'variationTypes', updatedVariations );
+	}
+
 	updateType( index, event ) {
-		event.preventDefault();
-		const updatedVariations = [ ...this.state.variations ];
+		const updatedVariations = [ ...this.props.product.variationTypes ];
 		updatedVariations[ index ] = { ...updatedVariations[ index ], type: event.target.value };
-		this.setState( { variations: updatedVariations } );
+		this.props.editProduct( this.props.product.id, 'variationTypes', updatedVariations );
 	}
 
 	updateValues( index, value ) {
```

Comment:
This pluralization confuses me. Is it a single value or multiple values? Below, `value` is assigned to `values:` which is equally confusing.